docs/source/VUDRM-token.md

```diff
@@ -0,0 +1,253 @@
+
+**VUDRM tokens should not be generated on the client side.**
+
+## VUDRM token structure
+
+```test-client|2018-26-11T11:01:04Z|cO9H1B2VKw0WyyNTOfoHEw==|05aff2dd06f4c52291b7a032c815ce8f599cf409```
+
+The VUDRM token is comprised of four components each separated by the pipe character (ASCII code 124):
+- The client name.
+- The time the token was generated in an ISO8601 format (yyyy-MM-ddThh:mm:ssZ).
+- The encrypted DRM policy.
+- A signed hash.
+
+## DRM policy
+
+The DRM policy is included in the VUDRM token as the encrypted third component. When using multiple DRM providers the parameters that are applicable to `ALL` will work across all DRM technologies. Parameters specific to one DRM provider can be used but will be ignored if not relevant. For example, a VUDRM token can be created that pertains to both Widevine and PlayReady. Any PlayReady specific parameters will only be applied when a PlayReady licence is required and will be ignored for Widevine.
+
+| Key                | Type     | Format                    | Applicable DRM | Description                                                                                                                     |
+|--------------------|----------|---------------------------|----------------|---------------------------------------------------------------------------------------------------------------------------------|
+| `contentid`        | string   | Any                       | ALL            | The content identifier. This is used for tracking content in the stats and is not required.                                     |
+| `polbegin`         | datetime | DD-MM-YYYY HH:mm:ss       | ALL            | The start date for the license. Default value depends on the DRM provider.                                                      |
+| `polend`           | datetime | DD-MM-YYYY HH:mm:ss       | ALL            | The end date for the license. Default value depends on the DRM provider.                                                        |
+| `liccache`         | string   | `yes` OR `no`             | ALL            | Should the license be cached. Implementation depends on the DRM provider.                                                       |
+| `firstplayback`    | integer  | Seconds                   | PlayReady      | Once playback initiated the user will have this window to complete playback. Once the window completes the license will expire. | 
+| `securitylevel`    | integer  | `150` OR `2000` OR `3000` | PlayReady      | https://docs.microsoft.com/en-us/playready/overview/security-level                                                              |
+| `type`             | string   | `r` OR `l` OR `p`         | Fairplay       | See [Fairplay DRM policy](fairplay-drm-policy) section for detail.                                                              |
```

Comment:
I believe all fairplay-drm-policy are missing section hash '#'. Like [Fairplay DRM policy](#fairplay-drm-policy)

Reply:
What do you mean?

Reply:
oh god... like this `[Fairplay DRM policy](#fairplay-drm-policy)`